src/Auth.ts

```diff
@@ -281,14 +281,24 @@ export class Auth {
   private ensureAccessTokenWithCertificate(resource: string, logger: Logger, debug: boolean): Promise<TokenResponse> {
     return new Promise<TokenResponse>((resolve: (tokenResponse: TokenResponse) => void, reject: (error: any) => void): void => {
       if (debug) {
-        logger.logToStderr(`Retrieving new access token using certificate (thumbprint ${this.service.thumbprint})...`);
+        logger.logToStderr(`Retrieving new access token using certificate...`);
       }
 
       let cert: string = '';
+      let thumbprint: string | undefined = this.service.thumbprint;
 
       if (this.service.password === undefined) {
         const buf = Buffer.from(this.service.certificate as string, 'base64');
         cert = buf.toString('utf8');
+
+        if (thumbprint === undefined) {
+          let pemObjs = pem.decode(cert);
```

Comment:
This can be `const` since it's not reassigned

Reply:
Argh 🙂